lib/mocha.js

```diff
@@ -145,14 +145,24 @@ Mocha.prototype.reporter = function(reporter, reporterOptions) {
     if (reporters[reporter]) {
       _reporter = reporters[reporter];
     }
-    // Try to load reporters from process.cwd() and node_modules
+    // Try to load reporters from process.cwd()
+    if (!_reporter) {
+      try {
+        _reporter = require(path.resolve(process.cwd(), reporter));
+      } catch (err) {
+        if (err.message.indexOf('Cannot find module') === -1) {
+          console.warn('"' + reporter + '" reporter blew up with error:\n' + err.stack);
+        }
+      }
+    }
+    // Try to load reporters from node_modules
     if (!_reporter) {
       try {
         _reporter = require(reporter);
       } catch (err) {
-        err.message.indexOf('Cannot find module') !== -1
-          ? console.warn('"' + reporter + '" reporter not found')
```

Comment:
This message seemed redundant in light of https://github.com/mochajs/mocha/pull/2453/files#diff-aa849a970cef551664c12f04a4209f6fL164